docs/articles/documentation/test-api/authentication/user-roles.md

```diff
@@ -0,0 +1,112 @@
+---
+layout: docs
+title: User Roles
+permalink: /documentation/test-api/authentication/user-roles.html
+---
+# User Roles
+
+Many test scenarios involve the activity of more than one user. TestCafe addresses these scenarios by providing a convenient way
+to isolate authentication test actions and apply them easily whenever you need to switch the user account.
+
+A piece of logic that logs in a particular user is called a *role*. Define a role for each user account participating in your test.
+
+## Creating and Using Roles
+
+Use the `Role` constructor to create and initialize a role.
+
+```text
+Role( func( t ) )
+```
+
+Parameter | Type     | Description
+--------- | -------- | --------------------------------------------------------------------------------
+`func`    | Function | An asynchronous function that contains logic that authenticates the user.
+`t`       | Object   | The [test controller](../test-code-structure.md#test-controller) used to access test run API.
+
+```js
+import { Role } from 'testcafe';
+
+const regularAccUser = Role(async t => {
+    await t
+        .typeText('#login', 'TestUser')
+        .typeText('#password', 'testpass')
+        .click('#sign-in');
+});
+
+const facebookAccUser = Role(async t => {
+    await t
+        .click('#sign-in-with-facebook')
+        .typeText('#email', 'testuser@mycompany.com')
+        .typeText('#pass', 'testpass')
+        .click('#submit');
+});
+
+const admin = Role(async t => {
+    await t
+        .typeText('#login', 'Admin')
+        .typeText('#password', 'adminpass')
+        .click('#sign-in');
+});
+```
+
+> Role initialization code is executed only once when the role is used for the first time.
+
+After you create the roles, you can switch between users at any moment except for the role initialization code.
```

Comment:
We need a couple of words about how switching happens: if role is not initialized yet, you will be navigated from the current page to Role login page and role initialization steps will be performed, then page from which you were navigated will be reloaded with new credentials. If role is already initialized current page will be reloaded with role credentials.